### ISSUE TITLE
Event before plugin change

### ISSUE THREAD

**Issue by Runemoro** [runelite/runelite#12731]
The `PluginChanged` event is only currently called after a plugin has been started or stopped. It would be nice to have a `BeforePluginChange` event that's called before the plugin's `startUp` or `shutDown` method is called.

This would make it possible to properly shut down my plugin before an incompatible plugin can start.

**Comment by Adam-** [comment]
This seems pretty hacky to me. What kind of incompatibilities do you have in your plugin?

**Comment by Runemoro** [comment]
If the GPU plugin starts, I want to shut down my plugin before it can start. Otherwise the game will crash because my plugin is already attached to the canvas when the GPU plugin tries to.

Maybe an "incompatible plugins" section in the plugin descriptor might be better, and have RL automatically shut down incompatible plugins before starting the other one?

**Comment by Adam-** [comment]
The *game* crashes? It's possible for plugins to throw out of startUp() on error, but that shouldn't be causing a game crash. Do you know why it's crashing?

**Comment by Nightfirecat** [comment]
Plugin conflict support was added to `@PluginDescriptor` in 24511aea4e50474768ab751e8214676af14e6dd8, which should resolve this issue.
